make/boards/tsb1.board

```diff
@@ -0,0 +1,69 @@
+# Define a board.
+#
+# A board describes a PCB with a microcontroller and peripherals (flash,
+# LEDs, sensors etc) which are used by this firmware build.
+
+# This is the TestSystemBoard1 tsb1
+
+# Determine board name and location (must be done before including any other files)
+BOARD_NAME          := $(notdir $(basename $(lastword $(MAKEFILE_LIST))))
+BOARD_MAKE_DIR      := $(patsubst %/,%,$(dir $(lastword $(MAKEFILE_LIST))))
+
+# Include the architecture definition
+$(call includeArch,efr32xg1x)
+
+MCU_PART            := EFR32MG12P432F1024GM48
+MCU_ARCH            := efr32xg1x
+MCU_FAMILY          := efr32xg12
+
+# Board options for compiler and linker, libraries
+CFLAGS      += -D$(MCU_PART)
+LDFLAGS     += -Wl,-T,$(SILABS_SDKDIR)/platform/Device/SiliconLabs/EFR32MG12P/Source/GCC/efr32mg12p.ld
+LDLIBS      += $(SILABS_SDKDIR)/platform/radio/rail_lib/autogen/librail_release/librail_efr32xg12_gcc_release.a
+
+INCLUDES    +=  \
+    -I$(SILABS_SDKDIR)/platform/radio/rail_lib/chip/efr32/efr32xg1x  \
+    -I$(SILABS_SDKDIR)/platform/Device/SiliconLabs/EFR32MG12P/Include \
+    -I$(SILABS_SDKDIR)/platform/CMSIS/Include
+
+SOURCES     += \
+    $(SILABS_SDKDIR)/platform/Device/SiliconLabs/EFR32MG12P/Source/GCC/startup_efr32mg12p.c \
+    $(SILABS_SDKDIR)/platform/Device/SiliconLabs/EFR32MG12P/Source/system_efr32mg12p.c
+
+# Pull in the board support code
+BOARD_SRC_DIR       := $(BOARD_MAKE_DIR)/../../src/boards/$(BOARD_NAME)
+INCLUDES            += -I$(BOARD_SRC_DIR)
+SOURCES             += $(BOARD_SRC_DIR)/platform.c
+
+# Pull in the platform_io.c
+ifndef NODE_PLATFORM_DIR
+$(error NODE_PLATFORM_DIR has not been defined!)
+endif
+SOURCES             += $(NODE_PLATFORM_DIR)/silabs/platform_io.c
+
+# Pull in default OS configuration, after possible platform-specific overrides
+ifneq ($(CMSIS_CONFIG_DIR),)
+INCLUDES            += -I$(CMSIS_CONFIG_DIR)
+endif
+
+# Allocate UUID for this board
+# NB! UNIQUE: MUST NOT BE USED IN OTHER BOARDS.
+# NB! CONST: CANNOT BE CHANGED FOR THIS BOARD.
+UUID_BOARD          := b1f61ed1-bc69-4850-9d09-1f388f554653
+
+# Maximum size limits for binaries in memory, their start addresses
+BOOTLOADER_START    := 0x00000
+BOOTLOADER_MAX_LEN  := 0x14000
+STAGE1_START        := 0x00000
+STAGE1_MAX_LEN      := 0x0C000
+STAGE2_START        := 0x0C000
+STAGE2_MAX_LEN      := 0x08000
+APP_START           ?= 0x14000
```

Comment:
Shall we have APP_START as 0 for tsb platform?

Reply:
Not necessarily, will want bootloader in some scenarios also.